src/store/model-store.js

```diff
@@ -49,10 +50,22 @@ class ModelStore extends Store {
     getModelByActionId(actionId) {
         return modelsByActionId[actionId];
     }
+
+
 }
 
 const store = new ModelStore();
 
+function deleteModelByAddress(address){
+    let result = readAddress(address);
+    if(result.type !== 'model' ) {
+        throw new Error('Address is not an model.');
+    }
+    delete models[result.index];
+    let itemsOfModel = ItemStore.getItems().filter(item => item.modelAddress === address);
+    itemsOfModel.forEach(item => ItemStore.deleteItemByAddress(item.address));
```

Comment:
Moved the deleteItemByAddress outside the store to its own function. ItemStore is now listing for DELETE_MODEL